src/contracts/storage/pool.rs

```diff
@@ -58,9 +55,14 @@ impl Default for Pool {
 }
 
 impl Pool {
-    pub fn create(init_tick: i32, current_timestamp: u64, fee_receiver: AccountId) -> Self {
+    pub fn create(
```

Comment:
It will be better to validate relation between `sqrt_price` and `init_tick` on storage level. The pool with desynchronized tick and price should not exist